search-replace.go

```diff
@@ -155,71 +162,229 @@ func fixLine(line *[]byte, replacements []*Replacement) *[]byte {
 	return line
 }
 
-var serializedStringPrefixRegexp = regexp.MustCompile(`s:(\d+):\\"`)
-
 func fixSerializedContent(line *[]byte, replacements []*Replacement) *[]byte {
-	startIndex := 0
-	for startIndex < len(*line) {
-		Debugf("Start of loop, startIndex: %d\n", startIndex)
-		match := serializedStringPrefixRegexp.FindSubmatchIndex((*line)[startIndex:])
-		if match == nil {
+	index := 0
+
+	var rebuiltLine []byte
+
+	for index < len(*line) {
+		Debugf("Start of loop, index: %d\n", index)
+		linePart := (*line)[index:]
+
+		details, err := parseEscapedData(linePart)
+
+		if err != nil {
+			if err.Error() == "could not find serialized string prefix" && index == 0 {
+				return line
+			}
+			// we've run out of things to parse, so just break out and append the rest
+			rebuiltLine = append(rebuiltLine, linePart...)
 			break
 		}
 
-		length, err := strconv.Atoi(string((*line)[startIndex+match[2] : startIndex+match[3]]))
-		if err != nil {
-			startIndex++
-			continue
+		rebuiltLine = append(rebuiltLine, (*line)[index:index+details.CurrentPartIndex]...)
+
+		index = index + details.NextPartIndex
+
+		content := linePart[details.ContentStartIndex : details.ContentEndIndex+1]
+
+		updatedContent := replaceInSerializedBytes(content, replacements)
+
+		// php needs the unescaped length, so let's unescape it and measure the length
+		contentLength := len(unescapeContent(updatedContent))
+
+		// but if the content never changed, we'll let the error be for safety.
+		if bytes.Equal(content, updatedContent) {
+			contentLength = details.OriginalByteSize
 		}
-		Debugf("Match found, length: %d\n", length)
 
-		contentStart := startIndex + match[1]
-		contentEnd := contentStart + length
+		// and we rebuild the string
+		rebuilt := "s:" + strconv.Itoa(contentLength) + ":\\\"" + string(updatedContent) + "\\\";"
+
+		rebuiltLine = append(rebuiltLine, []byte(rebuilt)...)
+	}
+
+	return &rebuiltLine
+}
+
+func replaceInSerializedBytes(serialized []byte, replacements []*Replacement) []byte {
+	for _, replacement := range replacements {
+		serialized = bytes.ReplaceAll(serialized, replacement.From, replacement.To)
+	}
+	return serialized
+}
+
+var serializedStringPrefixRegexp = regexp.MustCompile(`s:(\d+):`)
```

Comment:
```suggestion
var serializedStringPrefixRegexp = regexp.MustCompile(`s:(\d+):\\"`)
```
I'll fix this - I think I accidentally removed this.